### PR TITLE
fix: Better validate JSON double output

### PR DESCRIPTION
**Pull request by Swatinem**

### CODE REVIEW

File: src/sentry_json.c [hunk 1]
```diff
@@ -220,8 +221,16 @@ sentry__jsonwriter_write_double(sentry_jsonwriter_t *jw, double val)
     if (can_write_item(jw)) {
         char buf[50];
         // The MAX_SAFE_INTEGER is 9007199254740991, which has 16 digits
-        sentry__snprintf_c(buf, sizeof(buf), "%.16g", val);
-        write_str(jw, buf);
+        size_t written
+            = (size_t)sentry__snprintf_c(buf, sizeof(buf), "%.16g", val);
+        // print `null` if we have printf issues or a non-finite double, which
+        // can't be represented in JSON.
```
Comment (flub) [inline src/sentry_json.c:227]: doesn't this mean that it's impossible to create json adhering to some schema?  any consumer now has to expect null.  Might be better to error out entirely?

Comment (Swatinem) [reply src/sentry_json.c:227]: That is also true. Problem is that the codepaths right now are infallible (void functions), which means this is the only sensible things to do which at least does not produce outright invalid JSON.

Comment (mitsuhiko) [reply src/sentry_json.c:227]: Yeah, but at least that's consistent with what we do in other SDKs.

**Review by mitsuhiko — APPROVED** [review]

File: src/sentry_json.c [hunk 2]
```diff
@@ -220,8 +221,16 @@ sentry__jsonwriter_write_double(sentry_jsonwriter_t *jw, double val)
...
+        if (written < 0 || written >= sizeof(buf) || !isfinite(val)) {
+            write_str(jw, "null");
+        } else {
+            buf[written] = '\0';
```
Comment (mitsuhiko) [inline src/sentry_json.c:231]: Why is this necessary?

**Review by flub — APPROVED** [review]

File: src/sentry_json.c (hunk 2 above)
Comment (Swatinem) [reply src/sentry_json.c:231]: just to be safe. some invalid session payloads look like strings were not terminated correctly, and I remember reading somewhere that there are bugs around null-termination, also this here: https://stackoverflow.com/a/13067917
